guaranteed-message-delivery/README.md

```diff
@@ -0,0 +1,389 @@
+# Guaranteed Message Delivery
+
+## Overview
+
+In this tutorial, you will develop an integration that lets users reserve appointments at a hospital. To manage the flow of requests, you will employ a message broker. A publisher service will accept the appointment requests from users and publish them to a message broker. A consumer service will consume messages from the broker and interact with the hospital service to complete the reservation and send an SMS with the reservation status.
+
+To implement this use case, you will develop a REST service with a single resource using Visual Studio Code with the Ballerina Swan Lake extension. This resource will handle incoming user requests and forward them to the message broker. The consumer service that listens to the message broker's queue for new appointment requests will trigger a backend call to the hospital to make the reservation and send an SMS to the patient's phone number.
+
+The flow is as follows.
+
+1. The publisher service receives a request with a JSON payload in the following form.
+
+    ```json
+    {
+        "patient": {
+            "name": "John Doe",
+            "dob": "1940-03-19",
+            "ssn": "234-23-525",
+            "address": "California",
+            "phone": "+94710889877",
+            "email": "johndoe@gmail.com"
+        },
+        "doctor": "thomas collins",
+        "hospital_id": "grandoaks",
+        "hospital": "grand oak community hospital",
+        "appointment_date": "2023-10-02"
+    }
+    ```
+
+2. The publisher service publishes the received payload to the message broker.
```

Comment:
We can add something indicating responding to the original request here, to say the publisher bit ends there.